src/Fields/Checkbox.jsx

```diff
@@ -1,62 +1,18 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import Field from './Field';
+import createField from '../createField';
 
-export default class Checkbox extends Field {
-  static displayName = 'Field.Checkbox'
-
-  static propTypes = {
-    /* General */
-    id: PropTypes.string,
-    className: PropTypes.string,
-
-    /* Specific */
-    name: PropTypes.string.isRequired,
-    checked: PropTypes.bool,
-    disabled: PropTypes.bool,
-    required: PropTypes.bool,
-
-    /* Event handlers */
-    onFocus: PropTypes.func,
-    onChange: PropTypes.func,
-    onBlur: PropTypes.func
-  }
-
-  static defaultProps = {
-    expected: true,
-    checked: false,
-    disabled: false,
-    required: false
-  }
-
-  fieldWillRegister() {
-    const { checked } = this.props;
-
-    return {
-      ...this.props,
-      type: 'checkbox',
-      checked,
-      initialValue: checked
-    };
-  }
-
-  handleChange = (event) => {
-    const { contextProps } = this;
-    const { checked: nextValue } = event.currentTarget;
-    const prevValue = contextProps.get('checked');
-
-    return this.context.handleFieldChange({
-      event,
-      fieldProps: contextProps,
-      valueProp: 'checked',
-      prevValue,
-      nextValue
-    });
-  }
-
-  renderElement(props, contextProps) {
-    return (
-      <input checked={ contextProps.get('checked') } />
-    );
-  }
+function Checkbox(props) {
+  return (<input { ...props } />);
 }
+
+export default createField({
+  valueProp: 'checked',
+  mapPropsToField: props => ({
+    ...props,
```

Comment:
Is passing `...props` there needed?
I thought `return (<input { ...props } />);` Does the job?

Reply:
When you provide `mapPropsToField` you are expected to return *all* the refistration props, including default props.

I wouldn't add any default extension logic, this would only complicate things.

Reply:
What you may be getting wrong is where each props go.

`mapPropsToField` creates the record object in 'Form.state.field`.
`<input { ...props } />` destructs the composed props passed from the `connectField`'s `Field` class to the custom WrappedComponent.
`enforceProps` appends explicit props to the logic above.

We cannot access WrappedComponent.props in createField and Field class. The props can flow only down from Field to WrappedComponent. Form.state.field record is not related to props at all, that's a separate required object for field record management.